src/component/component/index.js

```diff
@@ -425,11 +424,11 @@ export class Component<P> extends BaseComponent<P> {
         }
     }
 
-    getRenderContext(element : ElementRefType) : string {
+    getRenderContext(context : ?string) : string {
 
-        if (element) {
```

Comment:
Is there a reason we're removing this check? If `element` is passed and both `iframe` and `popup` are availlable, we should always pick `iframe`

Reply:
I think the scenario you have mentioned may be redundant. When we remove the element check, this function will return `iframe`, if both `iframe` and `popup` are defined anyway, due to the following `if/else` statements. With the element check, if `iframe` is not a supported context, but `popup` is, an error will be thrown, even if the component has a valid `defaultContext` of `popup`, or the `init` function was called with `popup` as its context 

The problem with the element check is that this function was always validating `iframe`, regardless of how the component is configured for contexts, or what `defaultContext` is passed. This function was bypassed in favour of directly calling `validateRenderContext` in the `renderIframe` and `renderPopup` functions when instantiating a `ParentComponent`. 

I'm replacing the `element` check with a `context` check. If `iframe` and `popup` are available, and no default is defined, it will use `iframe` anyway as it is the first context to be checked. It makes the usage of this function more consistent throughout the code, and fixes the issue I've referenced with this pull request

Reply:
Makes sense. What about the following case; the component only allows `popup`, but `render()` is called with an element. In this case we should throw an error, right?

Reply:
An error may be a bit extreme in this case, I think. If an element reference is specified for a `popup`, it won't cause a `ParentComponent` to fail initialization. Maybe a warning would be more appropriate? I would have thought the `render` function is generic enough to instantiate either `context` with little change to the provided configuration.

Reply:
If we don't throw an error, we either have to:

1. Render to the element using an iframe, even though iframes are not allowed
2. Render to a popup and ignore the specified element

Neither of these are very intuitive for end-users. So it definitely sounds like an error-out case to me. Could we just pass the element through to `validateRenderContext`?

Reply:
That may be one solution, by giving the function 2 arguments.

I have a question though, is there any benefit to having a generic `render()` function in the first place? What functionality does it provide that explicit `renderIframe` and `renderPopup` functions don't have? Is there a particular use case for `render()`? Do you think it would be *more* intuitive to remove `render/renderTo` altogether, in favour of  having the end user call `renderIframe/renderIframeTo` and `renderPopup/renderPopupTo`? I feel it would be easier and clearer to define a more explicit API, rather than handling edge cases of a looser API.

Reply:
So, I guess if I had to go one direction or the other, I'd rather normalize on having only `render()` and `renderTo()` and allowing them to accept an enum value for the context, as an optional param.

The reason for that is, 90% of the use-cases for xcomponent involve writing a component that only works for one context -- it feels pretty rare to me that a component would be used as both a popup and an iframe. So I'd rather leave that in the hands of the component author rather than the component renderer. `renderIframe` and `renderPopup`, in retrospect, feel like an unnecessarily large api surface area.

Anyway, that's a discussion we can probably take outside this PR, if we need.

Reply:
Ok. So for this PR, I'll look into throwing an error if a component is configured for a `popup` context only, and an element reference has been specified

Reply:
Cool, sounds good 👍 happy to merge after that, everything else looks great